cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerModificationService.java

```diff
@@ -187,32 +187,40 @@ private Boolean isPrewarmed(Long clusterId) {
 
-    private void restartCMStaleServices(ClustersResourceApi clustersResourceApi) throws ApiException, CloudbreakException {
+    private void deployConfigAndRefreshCMStaleServices(ClustersResourceApi clustersResourceApi) throws ApiException, CloudbreakException {
         LOGGER.debug("Restarting stale services and redeploying client configurations in Cloudera Manager.");
         ServicesResourceApi servicesResourceApi = clouderaManagerApiFactory.getServicesResourceApi(apiClient);
         List<ApiService> services = servicesResourceApi.readServices(stack.getName(), SUMMARY).getItems();
         boolean configStale = services.stream().anyMatch(service -> !ApiConfigStalenessStatus.FRESH.equals(service.getConfigStalenessStatus()));
-        if (configStale) {
-            Optional<ApiCommand> optionalRestartCommand = clustersResourceApi.listActiveCommands(stack.getName(), SUMMARY).getItems().stream()
-                    .filter(cmd -> "RestartWaitingForStalenessSuccess".equals(cmd.getName())).findFirst();
-            ApiCommand restartServicesCommand;
-            if (optionalRestartCommand.isPresent()) {
-                restartServicesCommand = optionalRestartCommand.get();
-                LOGGER.debug("Restart for stale services is already running with id: [{}]", restartServicesCommand.getId());
-            } else {
-                restartServicesCommand = clustersResourceApi.restartCommand(stack.getName(),
-                        new ApiRestartClusterArgs().redeployClientConfiguration(Boolean.TRUE).restartOnlyStaleServices(Boolean.TRUE));
-            }
-            pollRestart(restartServicesCommand);
-            LOGGER.debug("Restarted stale services in Cloudera Manager.");
+        boolean clientConfigStale = services.stream().anyMatch(service -> !ApiConfigStalenessStatus.FRESH.equals(service.getClientConfigStalenessStatus()));
+        if (configStale || clientConfigStale) {
+            List<ApiCommand> commands = clustersResourceApi.listActiveCommands(stack.getName(), SUMMARY).getItems();
+            ApiCommand deployClientConfigCmd = getApiCommand(commands, "DeployClusterClientConfig", stack.getName(), clustersResourceApi::deployClientConfig);
+            pollDeployConfig(deployClientConfigCmd);
+            ApiCommand refreshServicesCmd = getApiCommand(commands, "RefreshCluster", stack.getName(), clustersResourceApi::refresh);
+            pollRefresh(refreshServicesCmd);
+            LOGGER.debug("Config deployed and stale services are refreshed in Cloudera Manager.");
         } else {
             LOGGER.debug("No stale services found in Cloudera Manager.");
         }
     }
 
+    private ApiCommand getApiCommand(List<ApiCommand> commands, String commandString, String clusterName, FunctionChecked<String, ApiCommand, ApiException> fn)
+            throws ApiException {
+        Optional<ApiCommand> optionalDeployClientConfigCommand = commands.stream().filter(cmd -> commandString.equals(cmd.getName())).findFirst();
```

Comment:
I think `optionalDeployClientConfigCommand` is a leftover name from refactor and should be `optionalApiCommand`, isn't it?

Reply:
yes, you are right, i will fix it